File: 2020_2021/L3_S5_2020-2021/INFO0503_modélisation web serv & web avancé/INFO503/TP5_projet_1_Sofiane/Projet_1_MEDDOURI_VANSCOOR/Voiture.php
<?php

class Voiture implements JsonSerializable{
    private $numeroIdentification;
    private $modele;
    private $moteur;
    private $dateFabrication;
    private $couleur;
    private $options = array();

    // constructeur

    public function __construct($numeroIdentification,$modele,$moteur,$dateFabrication,$couleur,$options){
    	$this->numeroIdentification = $numeroIdentification;
        $this->modele = $modele;
        $this->moteur = $moteur;
        $this->dateFabrication = $dateFabrication;
        $this->couleur = $couleur;
        $this->options = $options;
    }

    

    // getters

    public function getNumeroIdentification(){
        return $this->numeroIdentification;
    }

    public function getMoteur() {
        return $this->moteur;
    }

    public function getModele(){
        return $this->modele;
    }

    public function getDateFabrication() {
        return $this->dateFabrication;
    }

    public function getCouleur() {
        return $this->couleur;
    }

    public function getOptions() {
        return $this->options;
    }

    public function getStringOptions():String{
        $o = "";
        /*
        for($i = 0; $i < 4;$i++){
            $o .= $this->options[$i]->getNomOptions() . ", ";
        }
        */
        foreach($this->options as $element){
            $o .= $element->getNomOptions() . ",";
        }
        return $o;
    }

    // methode toString

    public function __toString(){
        return "numeroIdentification : " . $this->numeroIdentification . " modele : " . $this->modele . " moteur : " . $this->moteur . " couleur : " . $this->couleur . " options : " . $this->getStringOptions();
    }

    public function jsonSerialize() : array{
        return ['numeroIdentification' => $this->numeroIdentification, 'modele'=> $this->modele, 'moteur' => $this->moteur, 'dateFabrication' => $this->dateFabrication, 'couleur' => $this->couleur, 'options'=> $this->getStringOptions()];
    }

    public static function FromJson($json){
        $table = json_decode($json);
        return new self($table->numeroIdentification,$table->modele,$table->moteur,$table->dateFabrication,$table->couleur,$table->options);
    }
    
}
?>

File: 2020_2021/L3_S5_2020-2021/INFO0503_modélisation web serv & web avancé/INFO503/TP1/src/nathan/constructeur/Voiture/Voiture.java
package nathan.constructeur.Voiture;

import java.util.Date;

import nathan.constructeur.Voiture.moteur.Moteur;

public class Voiture {
	private static int id = 0;
	private int idVoiture;
	private Moteur moteur;
	private Date dateFab;
	private Couleur couleur;
	private String[] options;
	private Modele modele;
	
	public Voiture(Modele mod, Couleur c, Moteur m) {
		this.idVoiture = id;
		Voiture.id++;
		this.dateFab = new Date();
		this.moteur = new Moteur(m);
		this.couleur = c;
		this.options = new String[] {""};
		this.modele = mod;
	}
	
	public Voiture(Modele mod, Couleur c, Moteur m, String options[]) {
		this.idVoiture = id;
		Voiture.id++;
		this.dateFab = new Date();
		this.moteur = new Moteur(m);
		this.couleur = c;
		this.options = options;
		this.modele = mod;
	}
	
	public Voiture(Voiture v) {
		this.idVoiture = id;
		Voiture.id++;
		this.dateFab = new Date();
		this.moteur = new Moteur(v.getMoteur());
		this.couleur = v.getCouleur();
		this.options = v.getOptions();
		this.modele = v.getModele();
	}
	
	public int getIdVoiture() {
		return idVoiture;
	}
	
	public Voiture getVoitureByID(int idVoiture) {
		return null;
	}
	
	public Moteur getMoteur() {
		return moteur;
	}
	
	public void setMoteur(Moteur moteur) {
		this.moteur = moteur;
	}
	
	public Date getDateFab() {
		return dateFab;
	}
	
	public Couleur getCouleur() {
		return couleur;
	}
	
	public void setCouleur(Couleur couleur) {
		this.couleur = couleur;
	}
	
	public String[] getOptions() {
		return options;
	}
	
	public String getOptionsListString() {
		String s = "";
		for(int i=0; i<this.options.length; i++) s += this.options[i] + ", ";
		return s;
	}
	
	public void setOptions(String[] options) {
		this.options = options;
	}
	
	public Modele getModele() {
		return modele;
	}
	
	public String toString() {
		return (" id : " + this.idVoiture + "\n Créé le : " + this.dateFab.toString() + "\n Modèle : " + this.modele.toString() 
		+ "\n Couleur : " + this.couleur + "\n Moteur : " + this.moteur.toString() + "\n Options : " + this.getOptionsListString());
	}
	
}
